src/System.Buffers.Primitives/System/Buffers/Buffer.cs

```diff
@@ -106,14 +106,20 @@ public BufferHandle Pin()
 
         public bool TryGetArray(out ArraySegment<T> buffer)
         {
-            if (!_owner.TryGetArray(out buffer))
+            if (_owner!=null && _owner.TryGetArray(out var segment))
+            {
+                buffer = new ArraySegment<T>(segment.Array, segment.Offset + _index, _length);
+                return true;
+            }
+
+            if (_array != null)
             {
-                if (_array == null) return false;
                 buffer = new ArraySegment<T>(_array, _index, _length);
                 return true;
             }
-            buffer = new ArraySegment<T>(buffer.Array, buffer.Offset + _index, _length);
-            return true;
+
+            buffer = default(ArraySegment<T>);
```

Comment:
Will there ever be a scenario where both _array and _owner are both null causing us to get here?